README.md

```diff
@@ -1,5 +1,127 @@
+
+## How to run on your machine
+
+### Install tools
+
+If you don't have them already, you'll need node, bower, and gulp to be installed globally on your machine:
+
+1. Install [node](https://nodejs.org/en/download/). This includes npm -- the node package manager.
+2. Install [bower](https://bower.io/) globally: `$ npm install bower -g`
+3. Install [gulp](http://gulpjs.com/) globally: `$ npm install gulp-cli -g`
+
+### Clone the project and install dependencies
+
+1. Clone the project to your machine and open the folder:
+	```
+	$ git clone https://github.com/predix-ui/predix-ui.github.io.git
+	$ cd predix-ui.github.io
+	```
+2. Install dependencies:
+
+	```
+	$ npm install
+	$ bower install
+	```
+
+### Developing locally
+
+As you are developing, you can see your changes by running `gulp serve` from your command line. This will build the project files, serve the site locally, and automatically refresh the site as you make changes.
+
+### Running a local build
+
+When you think you are ready to deploy your changes, you can build the site locally to test how the changes will behave in production. The localBuild task runs additional tasks that simulate how the site will be built for production. Many of these tasks take a significant amount of time to run, so you should only run this command once you want test that your changes are ready to push to production.
+
+To run a local build:
+  ```
+  $ gulp localBuild
+  $ cd build/default
+  $ npm serve
+  ```
+
+### Running a production build
+You should never run the `gulp prodBuild` task manually. This task is run on Travis as part of our CI/CD process when a change is pushed to production.
+
+## Build a new component
+
+### Developing the component
+1. Generate a basic component structure using the [generator-px-comp](https://github.com/predixdev/generator-px-comp) Yeoman generator.
+2. Write the code to create a new component in that folder. (Have fun! 🎉)
+3. Write tests for your component that are now passing. (Nice work! 💯)
+4. Publish your component on bower:
+	```
+	$ bower register px-COMPONENT-NAME https://github.com/PredixDev/px-COMPONENT-NAME.git
+	```
+
+### Adding interactive documentation
+
+All components should have an interactive documentation and demo page on the [Predix UI catalog](https://www.predix-ui.com). These documentation pages are built automatically by Travis CI for each component when a new release (i.e. a new tag) is pushed to Github.
+
+Before you begin, you'll need to install the following tools:
+
+1. [Ruby](https://www.ruby-lang.org/en/) version 1.9.3 or higher (this could be tricky, and the ruby build that shipped with your system might be too old)
+2. [Travis CLI ruby gem](https://github.com/travis-ci/travis.rb)
+
+You should also do the following before starting:
+
+1. Make sure you switch off of BLUESSO (or any proxied Wifi connection) onto Internet (or any open Wifi connection). You should also turn off your VPN, if you have one.
```

Comment:
I do all this while on the corporate ethernet - not sure this advice is universal.